### PR TITLE
Use GitHub actions for continuous integration (CI)

### PR DESCRIPTION
**Pull request by SimonFrings**
Bye bye Travis CI, you've served us well.
@clue and I are currently working on switching from Travis to GitHub actions in all ReactPHP repositories.

I've built a little workaround for some tests for legacy PHP. When using recursion there is always the risk of reaching a full stack which results in PHP failing hard. Therefore the `max_nesting_level` setting from Xdebug (to prevent reaching this point) was set to 100 for PHP Versions < 5.4 which resulted in failing test for legacy PHP. In 2015 this setting was changed to 256 for PHP Versions > 5.4. For more details look into [Xdebug#1100](https://bugs.xdebug.org/bug_view_page.php?bug_id=00001100)

This code supports running tests on PHP 8.

Builds on top of #243 and reactphp/reactphp#442.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  pull_request:
+
+jobs:
+  PHPUnit:
+    name: PHPUnit (PHP ${{ matrix.php }} on ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-18.04
```
Comment (clue) [inline .github/workflows/ci.yml:14]: I see the other components are running on `ubuntu-20.04`, any reason to use an older distro here?

File: tests/FunctionalConnectorTest.php [hunk 2]
```diff
@@ -40,6 +40,10 @@ public function connectionToTcpServerShouldSucceedWithLocalhost()
      */
     public function testConnectTwiceWithoutHappyEyeBallsOnlySendsSingleDnsQueryDueToLocalDnsCache()
     {
+        if (DIRECTORY_SEPARATOR === '\\' || defined('HHVM_VERSION')) {
+            $this->markTestSkipped();
```
Comment (clue) [inline tests/FunctionalConnectorTest.php:44]: Why is this skipped?

File: tests/IntegrationTest.php [hunk 3]
```diff
@@ -101,6 +101,10 @@ public function gettingPlaintextStuffFromEncryptedGoogleShouldNotWork()
 
     public function testConnectingFailsIfConnectorUsesInvalidDnsResolverAddress()
     {
+        if (PHP_OS === 'Darwin') {
+            $this->markTestSkipped('Skipped on macOS due to a bug in reactphp/dns (solved in reactphp/dns#171)');
```
Comment (clue) [inline tests/IntegrationTest.php:105]: LGTM. Adding a link here only for reference: https://github.com/reactphp/dns/pull/171

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SimonFrings) [reply .github/workflows/ci.yml]: No reason, just a mistake on my end.

File: tests/FunctionalConnectorTest.php (hunk 2 above)
Comment (SimonFrings) [reply tests/FunctionalConnectorTest.php]: if ((DIRECTORY_SEPARATOR === '\\' && PHP_VERSION_ID < 70000) || defined('HHVM_VERSION')) {
            $this->markTestSkipped('Not supported on windows for PHP versions < 7.0 and legacy HHVM');
        }

**Review by clue — APPROVED** [review]
@SimonFrings Thanks for the update, changes LGTM! :shipit:

**Review by WyriHaximus — APPROVED** [review]

File: tests/IntegrationTest.php (hunk 3 above)
Comment (clue) [reply tests/IntegrationTest.php:105]: For the reference: We might as well update this now that https://github.com/reactphp/dns/pull/171 has been released a month ago. I'm good with this either way. :+1:

**Review by jsor — APPROVED** [review]

**Pull request merged**